### DISCUSSION TITLE
Installing Browsershot to work with Laravel/Laradock on MacOS

### DISCUSSION THREAD

**Discussion by paulgodard** [spatie/browsershot#753] [Q&A]
My objective is simple.  I want to convert an html page to a PDF.  I think Browershot is the best solution but I struggle to install it properly.  I am using Laravel 9 with Laradock on Docker Desktop on my Mac Ventura.

I installed Browsershot, Puppeteer & chromium inside laradock in Terminal as follow :

     cd laradock
     docker-compose exec workspace bash
     cd project

     npm install -g puppeteer
     npm install -g chromium
     composer require spatie/browsershot

No error there but when I try to use browsershot in a Laravel controller, I get this error:

    Browsershot::html('testing')->save('/public/assets/pdfs/example.pdf')->noSandbox()->setNodeBinary('/usr/bin/node')->setNpmBinary('/usr/bin/npm');

or

    Browsershot::url('http://travelearn.test/sc/view/134')->save('/public/assets/pdfs/example.pdf');

The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/var/www/travelearn/vendor/spatie/browsershot/src/../bin/browser.js' '{"url":"file:\/\/\/tmp\/1646675856-0888310001688983574\/index.html","action":"pdf","options":{"path":"\/public\/assets\/pdfs\/example.pdf","args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false}}'" failed. Exit Code: 127(Command not found) Working directory: /var/www/travelearn/public Output: ================ Error Output: ================ sh: 1: npm: not found sh: 1: node: not found 

I really need some help... please.

**Comment by shihab-salesdock** [comment]
I am also facing the same issue when I am trying in my local Mac

**Comment by aliaswpeu** [comment]
I have the same issue, I think puppeteer needs to be installed in laradock's php-fpm container
